contracts/external/cw-payroll/src/state.rs

```diff
@@ -0,0 +1,395 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Decimal, Deps, OverflowError, Uint128};
+use cw_storage_plus::{Item, Map};
+
+use crate::ContractError;
+use cw_denom::{CheckedDenom, UncheckedDenom};
+
+use wynd_utils::Curve;
+
+#[cw_serde]
+pub struct UncheckedVestingParams {
+    pub recipient: String,
+    pub amount: Uint128,
+    pub denom: UncheckedDenom,
+    pub vesting_schedule: Curve,
+    pub title: Option<String>,
+    pub description: Option<String>,
+}
+
+#[cw_serde]
+pub struct CheckedVestingParams {
+    pub recipient: Addr,
+    pub amount: Uint128,
+    pub denom: CheckedDenom,
+    pub vesting_schedule: Curve,
+    pub title: Option<String>,
+    pub description: Option<String>,
+}
+
+impl UncheckedVestingParams {
+    pub fn into_checked(self, deps: Deps) -> Result<CheckedVestingParams, ContractError> {
+        // Check vesting schedule
+        self.assert_schedule_vests_amount()?;
+
+        // Check valid recipient address
+        let recipient = deps.api.addr_validate(&self.recipient)?;
+
+        // Check denom
+        let checked_denom = match self.denom {
+            UncheckedDenom::Native(denom) => UncheckedDenom::Native(denom).into_checked(deps)?,
+            UncheckedDenom::Cw20(addr) => UncheckedDenom::Cw20(addr).into_checked(deps)?,
+        };
```

Comment:
```suggestion
let checked_denom = self.denom.into_checked(deps)?;
```

are you worried about another denom type being added and not getting a compilation error here? if so, fine to leave the match statement imo. could also check that in a test.